Sources/ActorSingletonPlugin/ActorSingleton.swift

```diff
@@ -134,3 +52,57 @@ public enum AllocationStrategySettings {
         }
     }
 }
+
+// ==== ----------------------------------------------------------------------------------------------------------------
+// MARK: ActorSingletonRemoteCallInterceptor
+
+struct ActorSingletonRemoteCallInterceptor<Singleton: DistributedActor>: RemoteCallInterceptor where Singleton.ActorSystem == ClusterSystem {
+    let system: ClusterSystem
+    let proxy: ActorSingletonProxy<Singleton>
+
+    func interceptRemoteCall<Act, Err, Res>(
+        on actor: Act,
+        target: RemoteCallTarget,
+        invocation: inout ClusterSystem.InvocationEncoder,
+        throwing: Err.Type,
+        returning: Res.Type
+    ) async throws -> Res
+        where Act: DistributedActor,
+        Act.ID == ActorID,
+        Err: Error,
+        Res: Codable
+    {
+        // FIXME: better error handling
+        guard actor is Singleton else {
+            fatalError("Wrong interceptor")
+        }
```

Comment:
Hm... I don't mind the check, it feels like we should be able to guarantee by construction it is correct, but good to keep the check...

Reply:
Yeah, I find it weird having this check, and something is seriously wrong if we ever got here and actor is of a different type, but then it also doesn't feel right without the check. Dunno...I might remove this in the end.

Reply:
Keeping is fine, just in case™